src/charm.py

```diff
@@ -184,7 +163,7 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         self._update_pebble_layer(event)
 
     def _update_pebble_layer(self, event: HookEvent) -> None:
```

Comment:
nit: Rename this to `_holistic_handler` to make it consistent with the rest of our refactored charms.